lib/ansible/modules/identity/keycloak/keycloak_realm.py

```diff
@@ -0,0 +1,1062 @@
+        revoke_refresh_token=dict(type='bool', aliases=['revokeRefreshToken']),
+        smtp_server=dict(type='dict', aliases=['smtpServer'], options={
+            'auth': dict(type='bool'),
+            'envelopeFrom': dict(type='str'),
+            'from': dict(type='str', required=True),
+            'fromDisplayName': dict(type='str'),
+            'host': dict(type='str', required=True),
+            'password': dict(type='str', no_log=True),
+            'port': dict(type='int'),
+            'replyTo': dict(type='str'),
+            'replyToDisplayName': dict(type='str'),
+            'ssl': dict(type='bool'),
+            'starttls': dict(type='bool'),
+            'user': dict(type='str'),
+        }),
+        ssl_required=dict(type='str', aliases=['sslRequired'], choices=['none', 'all', 'external']),
+        sso_session_idle_timeout=dict(type='int', aliases=['ssoSessionIdleTimeout']),
+        sso_session_max_lifespan=dict(type='int', aliases=['ssoSessionMaxLifespan']),
+        supported_locales=dict(type='list', elements='str', aliases=['supportedLocales']),
+        verify_email=dict(type='bool', aliases=['verifyEmail']),
+        wait_increment_seconds=dict(type='int', aliases=['waitIncrementSeconds']),
+    )
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=([['name', 'realm']]))
+
+    result = dict(changed=False, msg='', diff={}, proposed={}, existing={}, end_state={})
+
+    # Obtain access token, initialize API
+    kc = KeycloakAPI(module)
+
+    state = module.params.get('state')
+    realm = module.params.get('name') if module.params.get('name') is not None else module.params.get('realm')
+
+    # convert module parameters to realm representation parameters (if they belong in there)
+    realm_params = [x for x in module.params
+                    if x not in list(keycloak_argument_spec().keys()) + ['state', 'name'] and
+                    module.params.get(x) is not None]
+
+    # If the I(realm) is not set, assume I(name) as the default.
+    if module.params.get('realm') is None:
+        realm_params['realm'] = realm
+
+    before_realm = kc.get_realm_by_name(realm)
+    before_realm = {} if before_realm is None else before_realm
+
+    # Build a proposed changeset from parameters given to this module
+    changeset = dict()
+
+    for realm_param in realm_params:
+        # lists in the Keycloak API are sorted
+        new_param_value = module.params.get(realm_param)
+        if isinstance(new_param_value, list):
+            try:
+                new_param_value = sorted(new_param_value)
+            except TypeError:
+                pass
+        changeset[camel(realm_param)] = new_param_value
+
+    # Whether creating or updating a client, take the before-state and merge the changeset into it
```

Comment:
copy-paste issue:  we should talk about **realm** object here, not a _client_. To be globally replaced in this file...

Reply:
Ouch, I knew comments would bite me at some point. Thanks for catching this :)